documentation/STRUCTURE.md

```diff
@@ -0,0 +1,295 @@
+   typescript]
+1. [Done] integrate landing page (will be provided by a designer), everything
+   else from the Fairlearn project repositories
+1. [Done] add style template from pandas
+1. [Currently using fairlearn.github.io] deploy through fairlearn.org
+    - check that all pages are reachable through fairlearn.org
+    - https for fairlearn.org (currently only http works)
+    - [Done] remove API doc from readthedocs
+1. [Done] ensure the navigation from homepage to the other sections works
+    - [Done] manual testing
+1. automated testing of navigation/broken links
+1. Add an example notebook to show how to use estimators from various packages
+   with our mitigation techniques.
+
+The repository structure should look similar to
+[what scikit-learn has](https://github.com/scikit-learn/scikit-learn/tree/master/doc):
+
+- top-level doc directory has all the ReST files for the webpage except for
+  API documentation and example notebooks
+- API documentation comes directly from the code documentation
+- example notebooks live in a separate top-level directory
+  (scikit-learn calls it `examples`) as python files
+
+### Less urgent
+
+- Switch to numpy doc format; benefits explained in
+  [this issue](https://github.com/fairlearn/fairlearn/issues/314);
+  definitely worthwhile, but not as urgent as other items that get the webpage
+  started. Proposed solution in that issue was to switch over piece-by-piece.
+
+## Development process
+
+We need to document all the processes around generating documentation.
+Specifically, we need to document how one can
+
+- build/generate the documentation and subsequently view it
+- develop example notebooks that end up in python files, but perhaps while
+  creating them using Jupyter
+- get documentation changes into the Fairlearn repository
+  (PR with CI generating documentation and storing it as artifacts)
+
+Document exactly which tools/plugins we recommend, e.g. VSCode extensions
+or Jupytext, etc.
+
+## Outstanding questions
+
+1. Do we want users to cite us in any way? See
+   [this example](https://scikit-learn.org/dev/about.html#citing-scikit-learn).
+1. Do we want user testimonials? It definitely provides credibility (assuming
+   users are willing)
+1. Do we want a "News" section? It could list recent updates such as new
+   versions (link to changelog), but also upcoming presentations, references
+   to conference papers, blog posts, etc.
+1. Do we want a blog? [Example: pandas](https://pandas.pydata.org/community/blog/)
+1. Do we want to highlight differences to other fairness toolkits anywhere?
+1. Do we want to have an "ecosystem" page where we mention our relationship
+   with other projects such as [InterpretML](https://github.com/interpretml)
+1. Should we have a glossary?
+   [Example: scikit-learn](https://scikit-learn.org/dev/glossary.html)
+1. Do we want any kind of website analytics to figure out how users interact
```

Comment:
@adrinjalali do you have any analytics for `scikit-learn` ? My first thoughts were that this is probably not something that open source projects do, but when I went to check a few I was surprised to find a bunch that do use Google Analytics, for example `numpy` is one of them apparently https://github.com/numpy/numpy/blob/066f7ce37261fe82ccf974ccbebb74774d76f723/doc/neps/nep-0028-website-redesign.rst#analytics

As I wrote in the PR itself, it's probably very useful for the educational resources (user guide, notebooks) to know how many people view it, where we lose people (could be an indication to redesign material) and how people navigate through the page. I want to be super careful about this, though, because this sounds like data collection (GDPR? Privacy?). Beyond that, if we end up collecting data this can't just be accessible to, say, MSFT contributors, but it should be accessible by the entire community.

Any thoughts?

Reply:
I think we have google analytics as well in sklearn, but I had to check and I didn't know we have it, and I've never looked at the data. I think the GDPR issues are handled by now through notifications?

I'd agree that the data should be available to the community, but it can be that it's available if they ask for it. Or at least I wouldn't mind it if that's the case.

In terms of privacy, that's why I have all the blockers I've got and no data gets sent to any analytics server from my side lol

What I mean is that I understand the value of those data, and I'm okay if we collect them in this project, but I also recommend people to protect themselves especially since it's not too hard for people to do so via installing one or more adons.

I wouldn't worry about this too much, but I also wouldn't rely on the data too much. Having user surveys such as the ones done by dask and pandas people is a better way of understanding what the users do and want from the library.

Reply:
Perhaps if we do start collecting data, the 'agree' page should contain links to the various plugs which can thwart the trackers :-)

Less tongue-in-cheek, we are particularly concerned about this because of the potentially contradictory message it sends, as compared to the purpose of the repo (Differential Privacy would be the one worse project to have a tracker quietly appear). Thank you @adrinjalali for your thoughts - we may have been too paranoid. And your point about how reliable the data will be is well taken - it will take some time for us to get to our first million users, and hence for \sqrt{n} to beat down the errors.

We would welcome further outside perspectives on adding tracking (how much, sharing the data etc.).

Reply:
I hadn't even heard of these before! Very interesting, thanks Adrin!

https://pandas.pydata.org/community/blog/2019-user-survey.html